greenlight/lib/api/prefs.py

```diff
@@ -2,109 +2,174 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this file,
 # You can obtain one at http://mozilla.org/MPL/2.0/.
 
+from marionette.errors import MarionetteException
+
 from ..base import BaseLib
 
 
-class DefaultPrefBranch(BaseLib):
+class Preferences(BaseLib):
     archive = {}
 
-    @classmethod
-    def _cast(cls, value):
-        """
-        Interpolate a preference from a string.
+    def clear_user_pref(self, pref_name):
+        """Clear a user set preference.
 
-        - integers will get cast to integers
-        - true/false will get cast to True/False
-        - anything enclosed in single quotes will be treated as a string with
-          the ''s removed from both sides
+        :param pref_name: The preference to clear
         """
+        with self.marionette.using_context('chrome'):
+            return self.marionette.execute_script("""
+              Cu.import("resource://gre/modules/Services.jsm");
+              let prefBranch = Services.prefs.QueryInterface(Ci.nsIPrefBranch);
 
-        if not isinstance(value, basestring):
-            return value  # no op
-        quote = "'"
-        if value == 'true':
-            return True
-        if value == 'false':
-            return False
-        try:
-            return int(value)
-        except ValueError:
-            pass
-        if value.startswith(quote) and value.endswith(quote):
-            value = value[1:-1]
-        return value
+              let pref_name = arguments[0];
+
+              if (prefBranch.prefHasUserValue(pref_name)) {
+                prefBranch.clearUserPref(pref_name);
+                return true;
+              }
+              else {
+                return false;
+              }
+            """, script_args=[pref_name])
 
-    def get_pref(self, pref):
+    def get_pref(self, pref_name, default_branch=False, interface=None):
         """
-        Retrive a preference.
+        Retrieve a preference.
 
-        :param pref: The preference to inspect,e.g
-                     browser.tabs.remote.autostart
-        :returns: The value of the specified pref.
+        :param pref_name: The preference name to retrieve the value from
+        :param default_branch: Read from the default branch (default False)
+        :param interface: Interface to use for the complex value (default None)
+                          Possible values are: nsILocalFile, nsISupportsString,
+                          and nsIPrefLocalizedString
+
+        :returns: The value of the specified preference.
         """
+        assert pref_name
+
+        # Bug 1118825 - None is causing an exception to be raised
+        interface = interface or ''
+
         with self.marionette.using_context('chrome'):
             value = self.marionette.execute_script("""
```

Comment:
This would be consistent with clear_user_pref if it was just "return self.marionette.execute_script(""""